pkg/library/utils/k8s_utils.go

```diff
@@ -154,17 +159,16 @@ func GetLabel(obj metav1.Object, key string) string {
 	return obj.GetLabels()[key]
 }
 
-func GetClusterUID(ctx context.Context, c client.Client) (string, error) {
+func GetClusterUID(ctx context.Context, c dynamic.Interface) (string, error) {
 	//Already calculated? return it
 	if clusterUID != "" {
 		return clusterUID, nil
 	}
 
-	ns := &corev1.Namespace{}
-	err := c.Get(ctx, client.ObjectKey{Name: clusterIDNamespace}, ns)
+	un, err := c.Resource(corev1.SchemeGroupVersion.WithResource("namespaces")).Get(ctx, clusterIDNamespace, metav1.GetOptions{})
```

Comment:
Fair point, but i don't really think we want to add this to the topology. 

I was actually wondering when i was having to update this, if we even need it any more. It was required when we supported ACM that we had a consistent unique id per cluster and using this namespace id was determined to be the best way to do that. I'm not really sure if this is really required any longer though since ownership of records(in the provider) now comes down to the gateway/listener/dnsrecord combination and really is nothing to do with clusters (You could have many on the same cluster contributing to the same record set).

I don't want to change any of this as part of this PR, this is how it works currently, but a follow on task to look into the need for this any longer might be worthwhile.

@maleck13 fyi

Reply:
Ya that is fair. I think the less we change how it previously worked the better. Should there be a ticket created to address this and possible other things that were required because of the ACM support?

Reply:
We use this ID to ensure records in the DNSProvider don't conflict with each other in the multicluster context. I'm wondering if we can make topologies unique when creating them (we do client requests at that point as far as I understand) to booth be in a line with the rest policies and make sure we can differentiate from other clusters in providers :thinking:

Reply:
https://github.com/Kuadrant/kuadrant-operator/issues/959